src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -449,12 +474,20 @@ else if (vectorValues.size() < 1024)
         // evict from memory ASAP so better to do the PQ build (in parallel) one at a time
         synchronized (CassandraOnHeapGraph.class)
         {
-            if (vectorValues.dimension() >= 1536)
-                compressor = BinaryQuantization.compute(vectorValues);
+            if (compressedVectors instanceof PQVectors)
+            {
+                compressor = compressedVectors.getCompressor();
+                encoded = ((ProductQuantization) compressor).refine(vectorValues);
```

Comment:
compressor is a PQ object so you need to use it the same way as below when you get PQ from compute()